node/narwhal/src/helpers/committee.rs

```diff
@@ -324,118 +324,174 @@ pub mod prop_tests {
+            let committee = to_committee((u64::default(), validators.clone())).unwrap();
+            Self(committee, validators)
         }
     }
 
-    impl Validator {
-        pub fn get_account(&self) -> Account<CurrentNetwork> {
-            match Account::new(&mut rand_chacha::ChaChaRng::seed_from_u64(self.account_seed)) {
-                Ok(account) => account,
-                Err(err) => panic!("Failed to create account {err}"),
-            }
+    #[derive(Debug, Clone)]
+    pub struct ValidatorSet(pub HashSet<Validator>);
+
+    impl Default for ValidatorSet {
+        fn default() -> Self {
+            ValidatorSet(
+                (0..=4u64)
+                    .map(|i| Validator {
+                        account: Account::new(&mut rand_chacha::ChaChaRng::seed_from_u64(i)).unwrap(),
+                        stake: MIN_STAKE,
+                    })
+                    .collect(),
+            )
         }
+    }
+
+    impl Arbitrary for ValidatorSet {
+        type Parameters = ();
+        type Strategy = BoxedStrategy<ValidatorSet>;
 
-        pub fn is_valid(&self) -> bool {
-            self.stake >= MIN_STAKE
+        fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+            validator_set(any_valid_validator(), size_range(4..=MAX_COMMITTEE_SIZE as usize)).boxed()
         }
     }
 
-    impl CommitteeInput {
-        pub fn to_committee(&self) -> Result<Committee<CurrentNetwork>> {
-            let mut index_map = IndexMap::new();
-            for validator in self.validators.iter() {
-                index_map.insert(validator.get_account().address(), validator.stake);
-            }
-            Committee::new(self.round, index_map)
+    impl Arbitrary for CommitteeContext {
+        type Parameters = ValidatorSet;
+        type Strategy = BoxedStrategy<CommitteeContext>;
+
+        fn arbitrary() -> Self::Strategy {
+            any::<ValidatorSet>()
+                .prop_map(|validators| CommitteeContext(to_committee((1, validators.clone())).unwrap(), validators))
+                .boxed()
+        }
+
+        fn arbitrary_with(validator_set: Self::Parameters) -> Self::Strategy {
+            Just(validator_set)
+                .prop_map(|validators| CommitteeContext(to_committee((1, validators.clone())).unwrap(), validators))
+                .boxed()
+        }
+    }
+
+    pub fn any_valid_validator() -> BoxedStrategy<Validator> {
+        (MIN_STAKE..5_000_000_000, any_valid_account())
+            .prop_map(|(stake, account)| Validator { stake, account })
+            .boxed()
+    }
+
+    fn invalid_stake_validator() -> BoxedStrategy<Validator> {
+        (0..MIN_STAKE, any_valid_account()).prop_map(|(stake, account)| Validator { stake, account }).boxed()
+    }
+
+    pub fn any_valid_account() -> BoxedStrategy<Account<CurrentNetwork>> {
+        any::<u64>()
+            .prop_map(|seed| match Account::new(&mut rand_chacha::ChaChaRng::seed_from_u64(seed)) {
```

Comment:
I would've used `TestRng::fixed` otherwise, but it prints out the seed everytime, which in proptesting results in tons of spam in the test output, especially if shrinking.

Reply:
hmm, in that case perhaps we could use [XorShiftRng](https://docs.rs/rand_xorshift/0.3.0/rand_xorshift/struct.XorShiftRng.html) instead?

Reply:
Good idea. Let's do that in a separate PR.